gcp_variant_transforms/general_process.py

```diff
@@ -0,0 +1,151 @@
+# limitations under the License.
+
+"""General functions that are used by both vcf_to_bq and preprocessor.
+
+It includes parsing the command line arguments, reading the input, applying the
+PTransforms and writing the output.
+"""
+
+from argparse import ArgumentParser
+from argparse import Namespace  # pylint: disable=unused-import
+from typing import List  # pylint: disable=unused-import
+import enum
+
+import apache_beam as beam
+from apache_beam.io.filesystems import FileSystems
+from apache_beam import Pipeline  # pylint: disable=unused-import
+from apache_beam.pvalue import PCollection  # pylint: disable=unused-import
+
+from gcp_variant_transforms.beam_io import vcfio
+from gcp_variant_transforms.beam_io import vcf_header_io
+from gcp_variant_transforms.options.variant_transform_options import VariantTransformsOptions  # pylint: disable=unused-import
+from gcp_variant_transforms.transforms import filter_variants
+from gcp_variant_transforms.transforms import infer_undefined_headers
+from gcp_variant_transforms.transforms.merge_headers import MergeHeaders  # pylint: disable=unused-import
+
+# If the # of files matching the input file_pattern exceeds this value, then
+# headers will be merged in beam.
+_SMALL_DATA_THRESHOLD = 100
+_LARGE_DATA_THRESHOLD = 50000
+
+
+class PipelineModes(enum.Enum):
+  """An Enum specifying the mode of the pipeline based on the data size."""
+  SMALL = 0
+  MEDIUM = 1
+  LARGE = 2
+
+
+def parse_args(argv, command_line_options):
+  # type: (List[str], List[type]) -> (Namespace, List[str])
+  """Parses the arguments.
+
+  Args:
+    argv: A list of string representing the pipeline arguments.
+    command_line_options: A list of type ``VariantTransformsOptions`` that
+      specifies the options that will be added to parser.
+  """
+  parser = ArgumentParser()
+  parser.register('type', 'bool', lambda v: v.lower() == 'true')
+  options = [option() for option in command_line_options]
+  _add_parser_arguments(options, parser)
+  known_args, pipeline_args = parser.parse_known_args(argv)
+  _validate_args(options, known_args)
+  return known_args, pipeline_args
+
+
+def get_pipeline_mode(known_args):
+  # type: (Namespace) -> int
+  """Returns the mode the pipeline should operate in based on input size."""
+  if known_args.optimize_for_large_inputs:
+    return PipelineModes.LARGE
+
+  match_results = FileSystems.match([known_args.input_pattern])
+  if not match_results:
+    raise ValueError('No files matched input_pattern: {}'.format(
+        known_args.input_pattern))
+
+  total_files = len(match_results[0].metadata_list)
+  if total_files > _LARGE_DATA_THRESHOLD:
+    return PipelineModes.LARGE
+  elif total_files > _SMALL_DATA_THRESHOLD:
+    return PipelineModes.MEDIUM
+
+  return PipelineModes.SMALL
+
+
+def _add_parser_arguments(options, parser):
+  # type: (List[VariantTransformsOptions], ArgumentParser) -> None
+  for transform_options in options:
```

Comment:
Now that we have these in a separate module, I think it's ok to just inline them in the `parse_args` method (it becomes more readable that way).